code/components/gta-streaming-five/src/GameEventHooks.cpp

```diff
@@ -62,18 +107,144 @@ void* HandleEventWrap(void* group, rage::fwEvent* event)
 		}
 		catch (std::exception& e)
 		{
-		
 		}
 	}
 
 	return (*TFunc)(group, event);
 }
 
-#include <sstream>
+template<decltype(g_eventCall1)* TFunc>
+void* HandleEventWrapReact(rage::fwEventGroup* group, rage::fwEvent* event)
+{
+	if (event)
+	{
+		/*
+		 * event:
+		 * These events are stored with different member variables and with different layouts per class/struct,
+		 * so we have to investigate them case by case.
+		 * 
+		 * group:
+		 * In this reacting version the entity that triggers this event is stored in group->entities[17]
+		 * as well as in group->entities[18].
+		 */
 
-namespace rage
+		try
+		{
+			msgpack::sbuffer buf;
+			msgpack::packer packer(buf);
+
+			const char* eventName = typeid(*event).name() + 6;
+
+			packer.pack_array(4); // we'll offer 4 parameters
+			msgpack_pack(packer, eventName, event->GetId(), GetGuidFromBaseSafe(group->entities[17]));
+
+			// Quick 4 bytes check, i.e.: no strcmp
+			// this will turn the first 4 bytes after "CEvent" into an uint32 and check against that value
+			switch ((uint32_t&)eventName[6])
+			{
+				case ct_uint32("Shoc"): // Shocking
+				{
+					msgpack_array(packer, GetGuidFromBaseSafe((rage::fwEntity*&)event[13]), (float(&)[3])event[8]);
+					break;
+				}
+				case ct_uint32("Pote"): // Potential
+				{
+					// TODO: check how we will do this without a strcmp
```

Comment:
Same here. The entire function is filled with potentially slow code paths already, you won't save much of relevance with your jump table compare vs. a fixed-length (you know these names will not go out of the game EXE, so memcmp is fine) compare, especially compared to the impact on code complexity.

Reply:
I was assuming it would never leaves the EXE data, though never concluded it :)

Would you still opt to change both this and previous to an if branching memcmp version?

Reply:
Code complexity here is not worth the minimal gains.

Reply:
Aight, can't disagree